crates/ra_mbe/src/mbe_expander.rs

```diff
@@ -80,7 +85,7 @@ struct Bindings {
 #[derive(Debug)]
 enum Binding {
     Simple(tt::TokenTree),
-    Nested(Vec<Binding>),
+    Nested(FxHashMap<usize, Binding>),
```

Comment:
IIRC, `rustc` is using a `Vec` here: 

https://github.com/rust-lang/rust/blob/f843ad60ef5a78d8b4da85c5007d3ecac229f1a8/src/libsyntax/ext/tt/macro_parser.rs#L339-L359

I think we should probably follow what rustc is doing there .

Reply:
The reason i changed that is, when expanding, there is possible the index is not missing : 
Imagine this case :
```
macro_rules foo! {
  ($( ($($a:ident)* => $($b:ident))*))* => { ... }
}
foo!( ( => a) (b => ) )
```
Let me read how rustc handle that case first, but i suspect that using another method i don't know :)

Reply:
Okay, i got it now, `rustc` implementation push a empty bindings in it if there are zero match repeat. I will follow that.